### DISCUSSION TITLE
[Feature] Delete from web and sync to mobile

### DISCUSSION THREAD

**Discussion by rantanlan** [immich-app/immich#3594] [Feature Request]
### The feature

Tried to search for a feature I would love to see, seems no one mentioned it yet?

The abiliy to delete a photo in the web frontend and it also get's deleted on the synced phones... often I take pictures of notes, reminders, several picture of the same subject so sort them out on a big screen. Currently if you delete a synced file, it will show up in the webfrontend again after the next sync so I'm not really sure what the delete button on the web currently does.. (besides deleting photos which are not on your phone anymore) 

Or do I miss something?

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by marianjanco** [comment]
Hello, I would also appreciate this function. I am a hobby photographer and my common real workflow is like this:
1. take pictures with camera (let's say on a vacation)
2. transfer them through wifi to my phone
3. immich app synchronizes them to my server at home
4. when I come home, I want to revisit the pictures with the large screen and definitely delete bad ones on all devices (server + mobile phone)

It could also be generalized that pictures shouldn't be taken with a camera, but also with a mobile phone itself.

**Comment by dgsardina** [comment]
I  do believe that this feature is one of the most important ones. If I decide to delete a picture from the server, is needs to be deleted in the phone. Specially now that there is no way to delete a photo from both server and mobile at the same time

**Reply by NicholasFlamy** [reply]
> Specially now that there is no way to delete a photo from both server and mobile at the same time

As of a while ago this is possible on the mobile app. Obviously not on the web like this feature request asks for but just putting this here to let you know.

**Comment by YKhovanskikh** [comment]
Hello. Yeah, it's quite frustrating. After deleting files from the server, the phone begins transferring them back to the server. If this issue is a result of system limitations (iOS, Android), perhaps you could create a workaround? 
For instance:
- After deleting an image, marking it as deleted in DB (similar to the current recycle bin, but without restrictions on storage time)
- During synchronization with clients who have these photos saved locally, categorize them in a separate folder (category) 
- Users can enter this folder and manually clear it, with all necessary warnings from the system.

This is not fully user-friendly but more convenient than it is currently.

**Comment by theverex** [comment]
Would love this feature for all of the reasons listed above - I have tens of thousands of photos on my phone from over the years with no organisation. I would love to be able to organise & delete from Immich which then syncs back to my phone to delete them there as well & sync the albums. A much needed feature for me and I think it would be the last thing I need for Immich to be perfect!

**Comment by AZweimiller** [comment]
I am also looking for this feature. I am using Android and and I took a few test photos, waited for them to upload, and then deleted using the web. They disappeared from the mobile app timeline almost immediately. I checked my phone's DCIM folder using a file manager and the photos are still there. I expected them to be gone. They still show up in other gallery apps as well.

Consider this use case:

I will spam the shutter button on my camera to get a "keeper" (non-blurry) photo. This is especially important for kids who don't stop moving for photos. 😄 I will often sit down at the computer and prune the duplicates. I've enjoyed how Google Photos would remove the images from my mobile device. It helps reclaim a good bit of internal storage.

Thank you.

**Comment by markussiebert** [comment]
My use case is similar but different. As the sharing feature is crap at the moment, my wife and I use the same account. I would like to "remote delete" fotos on my wifes phone. They should not instantly get deleted - I think each client should as wether the files trashed should be deleted also on the client, or reuploaded (and if you confirm reupload, they should get restored from trash).

**Comment by benmccann** [comment]
The default gallery app gets permission to the file system whereas other apps can only access after a user interaction. That means Google Photos works differently for deletes on Pixels (where it's the default) and non-Pixels (where it normally isn't). You can't change the default gallery app without invoking ADB. This info is from https://www.androidpolice.com/2021/02/03/how-to-get-rid-of-google-photos-out-of-sync-changes-prompt-on-android-11/

That means that we have one of two choices on Android:
- [request access to manage all files](https://developer.android.com/training/data-storage/manage-all-files). This is discouraged, but allowed for things like backup and restore apps, which fits our usecase. It seems that we currently have these permissions [via a legacy flag](https://github.com/immich-app/immich/blob/9e122764e7e795addd8c6d266806db78dec0b815/mobile/android/app/src/main/AndroidManifest.xml#L4), so that's probably enough to allow this.
- emulate the workflow of Google Photos on non-Pixel phones, which looks something like this: https://github.com/Dilanwizzy/immich/blob/how-google-deletes-assets/HowGoogleDeletsImages.md. We can perhaps document how to connect ADB to remove that user interaction step for advanced users if we go this route

On iOS, it uses the non-Pixel behavior

Update 2/19/2025: the maintainers have decided that the review screen should occur on all devices by default, but Android can have an option to do automatic syncing

**Reply by benmccann** [reply]
The server sends an event via web socket when the mobile app is opened and otherwise you utilize the sync mechanism in `sync.service.dart`. `on_asset_trash` event when an item is sent to trash and an `on_asset_delete` event when an item is deleted. Currently when the Immich mobile application receives these events from the server, it only deletes the item from the local database, but doesn't delete them from the file system.

On Android, when a photo is being deleted here, also delete it from the file system:
https://github.com/immich-app/immich/blob/b21572cb32248fe2865a980025cebf7fc479ee29/mobile/lib/services/sync.service.dart#L191

To convert from ID to file path there, see this example:
https://github.com/immich-app/immich/blob/6b6d2a6621ff68e342a3879af274f18b860ee7ce/mobile/lib/services/search.service.dart#L106

On Android, for trashing a photo, make a new method `_handleOnAssetTrash` here: https://github.com/immich-app/immich/blob/b21572cb32248fe2865a980025cebf7fc479ee29/mobile/lib/providers/websocket.provider.dart#L162

When we get a trash event on android, rename the file in the manner that android does. I.e. .trashed-1720292136-PXL_20240606_185506468.jpg, which is the original photo name prepended with .trashed followed by a timestamp 30 days in the future (the time at which the photo should be removed from the trash and permanently deleted)

To test, upload some photos, delete them using the web application, and verify that they are then renamed to trashed or deleted on the file system after your change, so that they no longer appear in Google Photos: https://github.com/immich-app/immich/discussions/3594#discussioncomment-9110032

**Reply by baco7** [reply]
Are you saying this is how it should work with the current version?  It doesn't work this way for me.  When I use the server to delete a photo it doesn't show up in mobile app anymore but it remains in the file system and the filename doesn't change.  When I empty the trash on the server it reappears in the mobile app then re-uploads to the server.

**Reply by NicholasFlamy** [reply]
> Are you saying this is how it should work with the current version? It doesn't work this way for me. When I use the server to delete a photo it doesn't show up in mobile app anymore but it remains in the file system and the filename doesn't change. When I empty the trash on the server it reappears in the mobile app then re-uploads to the server.

The behavior you experienced is intended.
What @benmccann described was how it could theoretically be done.
My guess is this request will be fulfilled closer to stable similar to the free up space on mobile option.

**Reply by baco7** [reply]
Ah ok thanks

**Comment by benmccann** [comment]
I am not having the photo re-uploaded to web after deleting it as some have described here. However, I was able to reproduce it with the following steps:
- I selected an album with 3 photos for backup on the mobile app and backed it up
- I then deleted one of those three photos from the web instance
- The app still says "3 assets" under the backup section when I'd expect it to be 2 now and I can use Google Photos to find the deleted photo on my phone still when I would expect it to have been deleted

**Reply by mamerrem** [reply]
did you delete the photos from another device than the one with the backub job running?

**Reply by benmccann** [reply]
Yes. The backup was on mobile and I did the delete via the web

**Comment by przemhb** [comment]
For me personally such a function is essential and a deal breaker at the same time. I'd like to know what's the status of the feature request - has it been accepted, added to roadmap or maybe it is already being implemented?

**Comment by kiwijunglist** [comment]
Thanks would be keen on this too.

I deleted a bunch of photos on the web and then my phone added them back again.

**Comment by kiwijunglist** [comment]
The work around is to archive photos you don't want any more in the web instance and favorite your best photos.

In the future when the feature is available you can delete all the archive photos.

**Reply by ghan1t** [reply]
I tried this but went a step further for a workaround:

1. On the web: move all unwanted photos into the archive
2. On mobile, go to Library -> Archive -> select all photos --> Delete -> Confirm to delete from device
3. On the web: empty trash

I also would like if immich would either auto delete on my device or prevent duplicate uploads.

**Comment by DanTulovsky** [comment]
In my case, I deleted photos on the webapp but I can still see them in ios app.  The photos are not actually on my phone.  And selecting one of the deleted ones shows me an empty gray screen.  How do I update the ios thumbnail view?

**Reply by DanTulovsky** [reply]
fwiw, logging out of the app, and back in, refreshes the view and no longer shows the deleted photos. Is this working as intended?

**Comment by chrisscotland** [comment]
Just to add that this kind of bidirectional sync would be an important feature for me too. The prompt approach that Google photos takes to 'review out of sync changes' seems appropriate as long as sufficient detail is then provided to accurately review what changes are proposed and which you would like to accept. This would help catch any accidental deletions etc

**Comment by MikeFez** [comment]
Lack of this feature is the only thing holding me back from using Immich. I've tried a variety of different services (paid and self-hosted) to back up photos from my iPhone. Similar to what others have mentioned, I take pictures of notes, serial numbers, etc. for temporary usage. All services (other than Apple's iCloud from what I can tell, for iOS devices at least), simply sync from device to server, and then that "link" between photos is forgotten. Unfortunately, this means when I do my rounds and delete recent "temporary" images from my phone's default photo app or other 3rd party utility, they remain on the synced service.

Realistically, I'd like to be able to delete a photo from my phone _or_ immich, with the two left in sync.

I understand that the synced photos on immich can be considered a backup, and that this adds a bit of "danger" to that concept. However, the app could certainly present a "preview" of _what_ would be deleted from _where_ when deviations between the two occur. Also, said deleted images could persist in a "X day trash" for reversal if needed.

**Reply by blublub** [reply]
The regular deletion policy for such services is as MikeFez discribed:

Take the picture
it gets uploaded to immich/server) - and from here is not linked anymore, so if u delete it on the device it still exists on the server and vice versa.

This should be the standard policy in immich too - currently if one deletes images via the immich mobile app this deletion is propagated to the server - that is unexpected and can lead to unwanted data loss as ppl aren't used to that kind of behavior.

I understand the OPs intention for experienced "power users" who know what they do, but this synced behavior should be an optional choice and not standard

**Reply by NicholasFlamy** [reply]
> currently if one deletes images via the immich mobile app this deletion is propagated to the server

There are actually 3 delete buttons on mobile, the default delete button while viewing the photo might do this, but try multiselection and see that there are multiple delete buttons.

**Reply by MikeFez** [reply]
Yep, to clarify my original point, I don't think that this "bidirectional sync" should necessarily be default behavior. Immich for many seems to be their backup solution. However, the existence of a web UI with the ability to delete causes a mismatch between the data Immich has received vs what exists on the source device itself.

My thoughts on this topic stem from iCloud Photos. I take a picture on my iPhone and it's synchronized with my other apple devices. I can delete a photo from any of those devices, and they are all synced to unify the deletion and what the current state of my photo library is considered to be. The problem with this approach is that it's only a backup solution in the sense that if I were to lose one of my devices, I can replace it with another to again unify the new device with my photo library. However, if I were to delete all my photos (or a software/server glitch occurs where my photos disappear), then that is considered the current state, and so my images are removed from all my devices.

For that reason, I don't think that bi-directional sync via Immich should automatically happen (though, the ability to toggle it on would be nice for some I'd assume). That said, I'd love the option to review the "diff" between Immich and the source library. Just some sort of review UI, with Immich able to indicate "Hey, these 13 images were deleted on my end, should I go ahead and remove them from your source library as well?", and the reverse for images deleted from the source library which still exist in Immich.

Back to iCloud Photos, this resolves the problem of wanting a backup solution while at the same time mirroring what the current state of my photo library is. If all my photos are somehow deleted from my device, Immich would be aware of this and optionally offer to delete them from it's own database as well. I'd opt not to in this situation, with Immich having saved the day by having a complete backup of my library that I could then restore from.

Having tried various alternatives (Google photos, nextcloud, synology, etc), this is a feature they all seem to lack, causing the backups to contain an incredibly high number of images that I only intended to temporarily store. Some of them offer the ability to remove photos from my device if I were to delete the photos in their app, but none seem to do the reverse and detect deletions from my device which still exist within their service.

**Reply by benmccann** [reply]
You really should have an actual backup. E.g. you could create a disk image on a nightly or weekly basis locally and use something like Backblaze

I don't think there's any debate that when a photo is deleted from web it should be deleted from mobile as well. We simply haven't gotten around to implementing it yet. [I've been investigating it](https://github.com/immich-app/immich/discussions/3594#discussioncomment-9799150) though. Expect a PR soon

**Reply by MikeFez** [reply]
@benmccann I completely agree - I follow the 3-2-1 backup strategy for my household's data (Incremental snapshots across local & remote NAS, and a couple cloud services for specific items), however, it's the backup methodology for other people (non-techy users) that could would help simplify. Not that I think it's Immich's job to do this, but it seems to be capable & simple solution from a photo library-cloning perspective.

That's wonderful news regarding the events being in place, thanks for looking into this!

**Reply by blublub** [reply]
> I don't think there's any debate that when a photo is deleted from web it should be deleted from mobile as well. We simply haven't gotten around to implementing it yet. [I've been investigating it](https://github.com/immich-app/immich/discussions/3594#discussioncomment-9799150) though. Expect a PR soon

No it should not be deleted from a mobile device then - the mobile is NOT a mirror, it is an image source with usually less data space than a server and hence it will never be identical after some time has passed.

A delete, neither on mobile or server should be propagated to the other and vice versa in a "source to server model" **as the standard behavior.**

Changes in data on server should only be synchronized to server mirrors - so to a device which will always hold the same data as the server where the changes are applied on.

That is a basic model. 
I work in radiology, we have hundreds of millions of images on the server and we have images source (CT and MRI scanners etc) - it basically the same model as with immich just in the medical field.

Everyone would go nuts if a delete/change on the sources would trigger a delete on the server and vice versa - at some point in time this will end in absolute chaos as it would brake a "consistency of trust".
You are allowed to have any inconsistent "shit" on the source which will get deleted anyway at a watermark level but the server is the "holy grail of truth". 
The sources are a short term backup, if there is data loss at the server u can re-upload some days but thats it - the 3-2-1 rule has nothing to with synchronization of server and image sources! - it applies to a complete mirror of the server content / complete server backup, **that is independent to the storage on the source devices**

**Reply by simone-viozzi** [reply]
Sorry, I do not understand the point of viewing two different set of photos from web and mobile. The photo library is one, it should be the same everywhere you view it, no?

Then every change to a photo needs to be propagated to every device that hold that photo, no? 
and this of course should include deletion of a photo from the server.

Sorry, maybe it's me that I'm missing something here...

**Reply by blublub** [reply]
> Sorry, I do not understand the point of viewing two different set of photos from web and mobile. The photo library is one, it should be the same everywhere you view it, no?
> 
> Then every change to a photo needs to be propagated to every device that hold that photo, no? and this of course should include deletion of a photo from the server.
> 
> Sorry, maybe it's me that I'm missing something here...

Imagine u run out of storage space on your phone, so your 512gb phone storage is full.
When u delete images and video on the phone to free up space via the Immich app do u also want to delete them on the sever? - I guess not

As I explained in a previous post there is a whole industries who stores images long term in originating from multiple source and belive me they know why and there is a reason why deletions on end devices aren't synced to server and vice versa - but I guess just everyone on the internet knows better than GE Siemens Sectra Infinitt and Al the other companies...

**Reply by dgsardina** [reply]
> Imagine u run out of storage space on your phone, so your 512gb phone storage is full.
> When u delete images and video on the phone to free up space via the Immich app do u also want to delete them on the sever? - I guess not

I believe this feature request is aimed to exactly the opposite, to allow a way to delete the pics everywhere when on the web. Right now, the only way of deleting the pics on both phone and server is via the phone app, which is far from convenient.

**Reply by MikeFez** [reply]
> When u delete images and video on the phone to free up space via the Immich app do u also want to delete them on the sever? 

I use iCloud Photos (as many iOS users do now), so yes, I'd want them removed from the server since I never remove photos to free up space - I'm just removing them because I don't want them.

> I guess just everyone on the internet knows better than GE Siemens Sectra Infinitt and Al the other companies...

The issue here is that you're assuming the use case. The ability to delete from my phone is certainly an option that could at the very least be offered. With iCloud Photos, _none_ of my media are on my phone (other than a select portion); they all persist in iCloud storage, with portions being downloaded as I access them. The "photo library" contents on my phone are essentially just pointers to the cloud, which are downloaded on demand. 

If implemented neatly, I can envision a "Review" pane which shows which items would be removed from the server/phone, with people being able to selectively choose which ones they do want deleted.

Also, as @dgsardina mentioned, the feature being requested is the ability to detect what was deleted on the server in order to have it deleted from my phone. Realistically, I'd want it bi-directional though; with Immich mirroring my iCloud/mobile library.

**Reply by NicholasFlamy** [reply]
> Imagine u run out of storage space on your phone, so your 512gb phone storage is full. When u delete images and video on the phone to free up space via the Immich app do u also want to delete them on the sever? - I guess not

#165

Edit:
So deleting from mobile to free up space is a highly requested feature. So even if this discussion's sync deletes from web to mobile and/or vice-versa comes into effect, I would expect that it would be an option such that the current functionality is still an option.

**Comment by blublub** [comment]
Immich needs to be easy to use and to be fool proof.

This means that as the standard behavior a delete on web should not  delete the image on the phone and vice versa. 
The whole setup is a simple source to server model where where the source devices will always have less storage than the server. 

To avoid missing/lost images on the server and to have the ability to recover an image from the phone deletions should not be synced as the standard behavior. 

As I already mentioned this isn't a new problem and a whole industry doesn't sync deletion for good reasons. 

I don't care about An advanced feature where ppl set an option and know what they are doing - however most ppl when immich is stable won't know what they are doing and a fool proof method is to keep deletion on device and server seperate

**Reply by chrisscotland** [reply]
The way this is handled by gphotos seems to work well and cover a variety of use cases. A prompt on the phone when photos have been deleted from the web allows any changes to be reviewed and synced to the phone as desired. This way nothing is deleted by default but the option to remove photos on the phone exists. 

For me personally I use the web to review images full screen, check for focus and quality and delete any that I don't like. So personally for me if I delete on the web I would like that propagated to the phone in most cases. 

Removing images from the phone to free up storage space should not be called 'delete'. I think we should introduce different terminology for that use case where the images will remain on the server and use delete only where data will be removed permanently (from all sources).

**Reply by benmccann** [reply]
> As I already mentioned this isn't a new problem and a whole industry doesn't sync deletion for good reasons.

They absolutely do. If you delete a photo from Google photos web it will delete it on your phone.

> A prompt on the phone when photos have been deleted from the web allows any changes to be reviewed and synced to the phone as desired.

Note that Google Photos only does this when forced to. Google Photos doesn't do it on Pixel devices because they have the permissions to avoid it there as the default photo gallery

**Comment by srmncnk** [comment]
I would also very much appreciate this feature. I use web for culling and managing my albums, it's a magnificent tool for that!

They way syncing works however is not clear and it behaves differently when someone uploads to one server from multiple devices (which is my use case - multiple phones).

My suggestion is an option to sync one way (upload everything that has not been uploaded, don't touch local photos on phones). And add a date at when to start uploading (upload everything after a certain point of time - like Dropbox for instance).

**Reply by srmncnk** [reply]
FWIW, happy to help, flutter developer here. 👋

**Reply by benmccann** [reply]
> They way syncing works however is not clear and it behaves differently when someone uploads to one server from multiple devices (which is my use case - multiple phones).

Can you describe both how it behaves now and how you would expect it to behave?

> I would also very much appreciate this feature.
> My suggestion is an option to sync one way (upload everything that has not been uploaded, don't touch local photos on phones).

It sounds a bit like these two statements are in contradiction with each other. This ticket is saying that if you upload and item from your phone and then delete it on the web that it will also be deleted from your phone . You said you would appreciate the feature this issue describes. However, you then go on to say you want one way sync, which to me sounds like how it works today, so I'm not sure what you're describing when you say you want one way sync.

**Reply by srmncnk** [reply]
Right now it syncs everything. If file is deleted on web, it is uploaded from mobile again (not expected, I want to do my culling on web). If file is deleted on mobile and it's from that device, it's also deleted on the device (certainly did not expect that, also it's a bit confusing since I share the account with my wife and we both upload to the same account).
I expect that every file is uploaded that has yet not been uploaded (or have an option like dropbox to upload everything from now on). I don't except that files reupload when I delete them on web. I also don't expect that file gets deleted on device if I delete it on app in mobile (if it's mine).

TL;DR: I want to do culling on the web.

Sorry about the confusion. I want to stress out again that this is an amazing app and you all are doing a great job.

**Reply by benmccann** [reply]
> If file is deleted on web, it is uploaded from mobile again (not expected, I want to do my culling on web).

Yeah, I agree that should not happen.

>  If file is deleted on mobile and it's from that device, it's also deleted on the device (certainly did not expect that, also it's a bit confusing since I share the account with my wife and we both upload to the same account).

This part I disagree with. I think that's what should happen and it's how most other photo management systems work. E.g. if you delete a photo in Google Photos it will be removed from the file system. What's your use case for "deleting" a photo and leaving it on the file system? To me, that would make it very difficult to manage photos as you end up with orphaned files taking up storage space and you don't know which are ones you actually want and which aren't. Without knowing what you're trying to accomplish, would another solution like adding those photos to an album and hiding that album from the timeline work?

**Reply by srmncnk** [reply]
Fair enough, second part we can agree on. For culling, only the first part is relevant.

**Reply by benmccann** [reply]
> FWIW, happy to help, flutter developer here.

Would love to get some help on this. I did a pretty detailed write-up above of what would need to change (https://github.com/immich-app/immich/discussions/3594#discussioncomment-9036563). I think for Android the solution is quite simple and so maybe just implementing it there first would make sense. For iPhone, I haven't investigated as much how their file system permissions would work, but it sounds like Apple Photos users are used to a more complex UI than Pixel users are. Feel free to hop into the `#contributing` channel on Discord for questions or help getting started

**Reply by mfradcourt** [reply]
I'm using Synology Photo on web + iphone for a while now and my experience with the sync feature they implemented is quite smooth.
They have a section "Out-of-sync changes" on the iphone app, which list all photos/videos deleted from the server or moved to another space on the server. You can then choose yourself to delete those photos/videos from your phone.

I'm also currently testing immich and it is indeed surprising to have photos/videos that I deleted on the server being re-uploaded by the phone app (note that they of course re-upload only when emptying the trash bin on immich web).

It seems also that the behavior is a bit different if the immich app is open while deleting (and emptying trash) on the server versus if the immich app is closed. If the app is open, the photo is deleted immediately from the phone, if the app is closed then the photo will be considered to be uploaded again when the app is launched on the phone.

It gives the feeling that the phone app is the master and will always overwrite anything happening on the server if the app is not running, it this really what is expected ?

**Reply by bo0tzz** [reply]
> It seems also that the behavior is a bit different if the immich app is open while deleting (and emptying trash) on the server versus if the immich app is closed. If the app is open, the photo is deleted immediately from the phone, if the app is closed then the photo will be considered to be uploaded again when the app is launched on the phone.

When you delete on the web, the asset is moved into Immich's trash. The app also receives that event which is why you see it disappear from the timeline immediately. It's not actually deleting it from the device, and if you continue to fully delete it from the trash on the web you'll see it reappear on the mobile timeline as not uploaded.

**Comment by ciezczakmichal** [comment]
Today I deployed immich on server and it really lacks syncing of media deletion.

The only temporary solution I see is to use an external library and synchronize media using Syncthing, instead of using mobile app backup feature. Then regardless of where I delete a photo - whether on my phone, computer, or in the immich web app - the changes will be applied to other devices by Syncthing. From what I read in docs, the photo in immich app should end up in the trash when file is gone.

**Comment by dionorgua** [comment]
Same here. Just decided to try it and was confused that some photos that I deleted from web interface were automatically added again.

I see a few possible workarounds for now:
- Never empty Immich trash and Increase automatic cleanup to very big value
- Make Immich at keep list of 'deleted' photos forever. Just metadata (optionally with checksums). 

I think that second options safe enough. App still never deletes anything

**Comment by dionorgua** [comment]
Same here. Just decided to try it and was confused that some photos that I deleted from web interface were automatically added again.

I see a few possible workarounds for now:
- Never empty Immich trash and Increase automatic cleanup to very big value
- Make Immich at keep list of 'deleted' photos forever. Just metadata (optionally with checksums). 

I think that second options safe enough. App still never deletes anything

PS. I think it's even more risky to 'delete' files manually from native gallery app or other file manager

**Comment by aleksandrsovtan** [comment]
A PR has been created implementing the synchronization of local media files on Android with the remote server.

https://github.com/immich-app/immich/pull/16153

**Reply by vittorelli** [reply]
The PR hereabove has been closed in favor of #16732 where feedback has been addressed.

**Reply by svenstaro** [reply]
I believe this should now be fixed as of immich v1.132.3 and https://github.com/immich-app/immich/pull/16732. Maybe this feature request should now be closed?

**Reply by NicholasFlamy** [reply]
So, it's implemented on Android but not (yet?) on iOS.
I'm not sure if iOS is possible.

**Reply by svenstaro** [reply]
Oh, good point.

**Reply by srmncnk** [reply]
Can we at the very least delete from server and acknowledge it on mobile (not upload again)? This is one of the main reasons this feature request exists.

**Reply by benmccann** [reply]
We will work on adding an implementation for iOS, which will have a review screen allowing you to select which trash events to sync to mobile. This will also likely become the default on Android. The just merged implementation that automatically syncs deletes on Android will continue to be available at that point as an option for people who want to skip the review screen on Android.

The remaining work on this feature is temporarily on hold while the team works on migrating the mobile app to Sqlite as that will make implementing this feature easier.

**Reply by geforcefan** [reply]
@benmccann Is there any news on this (iOS)?

**Reply by benmccann** [reply]
Currently working on getting the Android code ported to the beta timeline: https://github.com/immich-app/immich/pull/20473

That's close to done and then work can switch to iOS.

**Comment by grid8** [comment]


**Reply by NicholasFlamy** [reply]
Implemented on Android Immich Beta but not iOS I think. Also, asking for updates is bothersome. We usually post updates.

**Comment by ARVEDz** [comment]
On iOS, I delete all the photos on web or app, but to make sure I delete them as well from my iCloud library I only empty the trash bin on my phone, so all the local pictures on my iOS Photo App get deleted as well. What is just kind of annoying, is that in the app I don't have an empty trash bin button like I have in the web. Which means I have to select all the monthly pictures separately.

Would be amazing if you could add such a "Empty Trash Bin" button in the iOS immich app too.

**Comment by foobarth** [comment]
Chiming in here as it's quite frustrating spending more than an hour of selecting hundreds of to-be-deleted photos synced from the mobile for deletion, just to see them reappear as soon as the trash is emptied. On top, as @ARVEDz said, there isn't even a proper empty-the-trash button on the iOS app, so you select again and again. 
This is really bad UX holding me back from actually switching to Immich. Google Photos, though i really want to get away from Google, manages deleted images flawlessly, no matter where they get deleted.

How about implementing it like so:
1. On each app or syncing client, before starting the sync, retrieve a list of (trashed or permanently) deleted images (hashes) from the server. This list must be kept also for permanently deleted images, at least as long as they are present on some device.
2. If photos in that list are still present on the device, either ask the user if they should be deleted on the device as well or just ignored, so they don't get re-uploaded again. This would keep the local copy outside of Immich available but it won't resync again.

Currently, i really don't know how to proper handle photo deletion in a way that they don't reappear.

**Comment by benmccann** [comment]
The PR that I've been sponsoring for an automatic opt-in version of this feature for the new timeline on Android was just approved. I expect that will be merged next week and be in the next release. Work is also underway for a version of this feature built on top of that with a review screen for synced deleted assets. That will be the default on Android and the only implementation on iOS.

**Comment by hTepAchtetepe** [comment]
Hello,

Why Immich won't work as Google photo.
- 1st sync, all is backed up
- if I take new pictures, new pictures are downloaded on the server and is reflected in immich iOS app
- If picture is deleted from web portal, it's deleted on the immich app (normal) but it's not deleted on my device camera roll. No change on future sync, immich db is ok
- if picture is deleted from iOS Immich app, prompt asks me if I want to delete the picture in the camera roll
- if picture is deleted from iOS camera roll, no pb, I have a backup on Immich, iCloud is another library maybe with mess 

Immich is a library, iCloud is another.

To be honest, I don't want immich delete picture on my device expect of if I operate from the immich app.

Google photo works like that, synology photos works like that.

**Reply by NicholasFlamy** [reply]
It sounds like it's working exactly the way you want it.

Edit: You didn't describe anything realted to this Feature Request. Also, I don't believe this Feature Request was implemented on iOS.

**Reply by hTepAchtetepe** [reply]
Not exactly from my own.
When I clean my immich on the web portal (duplicate or simple deleting), if some pictures are on my iOS device, pictures are newly downloaded so I can't clean these pictures.

Is my immich Brocken?

I'd like immich iOS app doesn't care of my past sync picture.

**Reply by benmccann** [reply]
This will be implemented.

Currently there is an option to make that happen automatically in Android. By default, on Android, when an asset is deleted you will be asked to review it: https://github.com/immich-app/immich/pull/23992

Once that PR is merged to finish up the Android implementation then it will be implemented for iOS next.

**Reply by hTepAchtetepe** [reply]
@benmccann ,
Great news...
Just for information, why are you extending sync from immich to iOS library?

**Reply by NicholasFlamy** [reply]
#28280 seems to replace the previous PR.

**Comment by aufruf** [comment]
I am also looking for this. Delete photos in web and remove those also from my iPhone. @benmccann : any news when this will be on the roadmap? And thanks for all the work, it’s a fantastic project.

**Comment by baek-sang** [comment]
https://github.com/immich-app/immich/discussions/4282

I don't know about everyone else, but in my case, features like image editing, slides, workflows, and many of the other extras in Immich are unnecessary.

I'm actually curious how many users make regular use of those features.

All I really need is for it to reliably back up the photos from my phone.

I'd also like deleted photos in the Immich web app to be removed from the iOS Photos app, and when a photo is deleted in the iOS Photos app, Immich should notify me that it's been deleted and let me choose whether to remove it from Immich as well.
